core/vtk/ttkOFFReader/ttkOFFReader.cpp

```diff
@@ -0,0 +1,136 @@
+#include "ttkOFFReader.h"
+
+#include "vtkCellType.h"
+#include "vtkDataObject.h"
+#include "vtkDoubleArray.h"
+#include "vtkInformation.h"
+#include "vtkInformationVector.h"
+#include "vtkObjectFactory.h"
+#include "vtkPointData.h"
+#include "vtkPoints.h"
+#include "vtkSmartPointer.h"
+#include "vtkStreamingDemandDrivenPipeline.h"
+#include "vtkUnstructuredGrid.h"
+
+#include <iostream>
+#include <sstream>
+
+vtkStandardNewMacro(ttkOFFReader);
+
+// Public
+// {{{
+
+void ttkOFFReader::PrintSelf(ostream &os, vtkIndent indent) {
+  this->Superclass::PrintSelf(os, indent);
+
+  os << indent << "File Name: " << (this->FileName ? this->FileName : "(none)")
+     << "\n";
+}
+
+// }}}
+// Protected
+// {{{
+
+ttkOFFReader::ttkOFFReader() {
+  this->FileName = NULL;
+  this->SetNumberOfInputPorts(0);
+  this->SetNumberOfOutputPorts(1);
+}
+
+int ttkOFFReader::RequestData(vtkInformation *request,
+                           vtkInformationVector **inputVector,
+                           vtkInformationVector *outputVector) {
+
+  ifstream offFile(FileName, ios::in);
+
+  if (!offFile) {
+    cerr << "[ttkOFFReader]: Can't read file: " << FileName << endl;
+    return -1;
+  }
+
+  std::string FileType;
+
+  offFile >> FileType;
+  if (FileType != "OFF") {
+    cerr << "[ttkOFFReader]: Bad format for file: " << FileName << endl;
+    return -2;
+  }
+
+  vtkSmartPointer<vtkPoints> points = vtkSmartPointer<vtkPoints>::New();
+  vtkSmartPointer<vtkUnstructuredGrid> mesh =
+      vtkSmartPointer<vtkUnstructuredGrid>::New();
+  vtkSmartPointer<vtkDoubleArray> scalars =
+      vtkSmartPointer<vtkDoubleArray>::New();
+
+  int nbVerts, nbFaces, nbEdges, curLine;
+  std::string line;
+
+  offFile >> nbVerts >> nbFaces >> nbEdges;
```

Comment:
I don't believe most OFF files provide the number of edges in the header (look at examples on the AIM@SHAPE repository for instance)

Reply:
This is fixed in commit 2dffc5a53fa9b52fa564ff0e874c7f6f6e0cf0e7